simulator/core/gdb.c

```diff
@@ -400,6 +400,7 @@ static bool _wait_for_gdb(void) {
 			// kill
 			INFO("Killed by remote debugger, dying\n");
 			sim_terminate(true);
+			break;
```

Comment:
Ahh... nice catch. One of those 'not technically possible' things – once upon a time, `sim_terminate` was marked `__attribute__ ((noreturn))` so this wasn't actually a fallthrough (and with the argument `true`, it still isn't, but the compiler can't know that). Later on I added an option to trap on termination to allow a debugger to hook and analyze the final core

Reply:
Odd. Usually GCC can catch things like this especially if `should_exit` is defined as a constant (which it is in this case). Maybe the C library `exit` isn't defined as noreturn?